### PR TITLE
Persist user category names during export/import

### PR DESCRIPTION
**Pull request by deXol**
- Persist user category names and isBle information in the exported payload
- Loading user category names during import if available
- Changed magic number in import functions

### CODE REVIEW

File: src/MPDevice.cpp [hunk 1]
```diff
@@ -4905,16 +4909,20 @@ bool MPDevice::readExportPayload(QJsonArray dataArray, QString &errorString)
 {
     /** Mooltiapp / Chrome App save file **/
 
+    const auto dataSize = dataArray.size();
+    const bool isBleExport = EXPORT_BLE_DATA_SIZE == dataArray.size() && dataArray[EXPORT_IS_BLE].toBool();
+    const QString deviceVersion = dataArray[EXPORT_DEVICE_VERSION].toString();
     /* Checks */
-    if (!((dataArray[9].toString() == "mooltipass" && dataArray.size() == 10) || (dataArray[9].toString() == "moolticute" && dataArray.size() == 14)))
+    if (!((deviceVersion == "mooltipass" && dataSize == EXPORT_MP_DATA_SIZE)
```
Comment (limpkin) [inline src/MPDevice.cpp]: EXPORT_MP_DATA_SIZE might be confusing here... Maybe rename is as "number of fields for moolticute export"?

File: src/MPDevice.cpp [hunk 2]
```diff
@@ -4905,16 +4909,20 @@ bool MPDevice::readExportPayload(QJsonArray dataArray, QString &errorString)
...
+          || (deviceVersion == "moolticute" && (dataSize == EXPORT_MC_DATA_SIZE || isBleExport))))
```
Comment (limpkin) [inline src/MPDevice.cpp]: same applies

File: src/MPDevice.cpp [hunk 3]
```diff
@@ -4923,22 +4931,22 @@ bool MPDevice::readExportPayload(QJsonArray dataArray, QString &errorString)
...
         qDebug() << "Imported data change number: " << importedDataDbChangeNumber;
-        importedDbMiniSerialNumber = dataArray[13].toInt();
+        importedDbMiniSerialNumber = dataArray[EXPORT_DB_MINI_SERIAL_NUM].toInt();
         qDebug() << "Imported mini serial number: " << importedDbMiniSerialNumber;
     }
 
     /* Read CTR */
     importedCtrValue = QByteArray();
-    auto qjobject = dataArray[0].toObject();
+    auto qjobject = dataArray[EXPORT_CTR].toObject();
```
Comment (limpkin) [inline src/MPDevice.cpp]: maybe append with _INDEX this identifiers?

Comment (deXol) [reply src/MPDevice.cpp]: should I keep EXPORT_ too? because with both some of them will be really long

**Comment by limpkin** [comment]
Why not.... We are in 2019 and have wide screens anyway no?

**Pull request merged**